### PR TITLE
OTG RT-1.3: BGP Route Propagation (Add WatchAll)

### LINKED ISSUE #1488: OTG RT-1.3: BGP Route Propagation GetAll returns  "Value not present"
**Describe the bug**
BGP Route Propagation test has GetAll which retrieves learnt prefixes. GetAll fails sometimes when BGP is taking sometime to exchange the routes. 

To address this issue, we need to add WatchAll with timer to check if routes are learnt and use GetAll

Failed Log:
```
=== RUN   TestBGP
=== RUN   TestBGP/propagate_IPv4_over_IPv4
    route_propagation_test.go:571: Advertise prefixes from ATE port1, observe received prefixes at ATE port2
    route_propagation_test.go:578: Configure Network Instance
    route_propagation_test.go:100:
        *** Creating new config for otg...


    route_propagation_test.go:182: Pushing config to ATE and starting protocols...
    route_propagation_test.go:183:
        *** Pushing config to otg...


    route_propagation_test.go:184:
        *** Starting protocols on otg...


    route_propagation_test.go:590: Verify DUT BGP sessions up
    route_propagation_test.go:238: BGP sessions established
    route_propagation_test.go:592: Verify OTG BGP sessions up
    route_propagation_test.go:598: Checking for BGP Prefix 198.51.100.0/32
    route_propagation_test.go:598: Checking for BGP Prefix 198.51.100.1/32
    route_propagation_test.go:598: Checking for BGP Prefix 198.51.100.2/32
    route_propagation_test.go:598: Checking for BGP Prefix 198.51.100.3/32
=== RUN   TestBGP/propagate_IPv6_over_IPv6
    route_propagation_test.go:571: Advertise IPv6 prefixes from ATE port1, observe received prefixes at ATE port2
    route_propagation_test.go:578: Configure Network Instance
    route_propagation_test.go:100:
        *** Creating new config for otg...


    route_propagation_test.go:182: Pushing config to ATE and starting protocols...
    route_propagation_test.go:183:
        *** Pushing config to otg...


    route_propagation_test.go:184:
        *** Starting protocols on otg...


    route_propagation_test.go:590: Verify DUT BGP sessions up
    route_propagation_test.go:238: BGP sessions established
    route_propagation_test.go:592: Verify OTG BGP sessions up
    route_propagation_test.go:611: Checking for BGP Prefix 2001:db8:1::1/128
    route_propagation_test.go:333: GetAll(t) on {ate: ATE{Dims:Dims{Name:otg Vendor:IXIA HardwareModel: SoftwareVersion: Ports:map[port1:Port{Name:eth1 Speed:SPEED_UNSPECIFIED CardModel: PMD:PMD_UNSPECIFIED} port2:Port{Name:eth2 Speed:SPEED_UNSPECIFIED CardModel: PMD:PMD_UNSPECIFIED}] CustomData:map[$KEY_SERVICE_MAP:map[gnmi:name:"gnmi" inside:50051 outside:50051 outside_ip:"172.18.0.50" inside_ip:"10.96.164.112" node_port:30781 grpc:name:"grpc" inside:40051 outside:40051 outside_ip:"172.18.0.51" inside_ip:"10.96.83.17" node_port:31877 https:name:"https" inside:8443 outside:8443 outside_ip:"172.18.0.53" inside_ip:"10.96.76.178" node_port:32524]]}}} at /bgp-peers/bgp-peer[name=port2.dev.BGP6.peer]/unicast-ipv6-prefixes/unicast-ipv6-prefix[address=*][origin=*][path-id=*][prefix-length=*]: query "/bgp-peers/bgp-peer[name=port2.dev.BGP6.peer]/unicast-ipv6-prefixes/unicast-ipv6-prefix[address=*][origin=*][path-id=*][prefix-length=*]": value not present
=== RUN   TestBGP/propagate_IPv4_over_IPv6
    route_propagation_test.go:571: IPv4 routes with an IPv6 next-hop when negotiating RFC5549 - validating that routes are accepted and advertised with the specified values.
    route_propagation_test.go:574: TODO: RFC5549 needs to be enabled explicitly and OpenConfig does not currently provide a signal.
--- FAIL: TestBGP (202.07s)
    --- PASS: TestBGP/propagate_IPv4_over_IPv4 (111.83s)
    --- FAIL: TestBGP/propagate_IPv6_over_IPv6 (90.24s)
    --- SKIP: TestBGP/propagate_IPv4_over_IPv6 (0.00s)
```

### PR DESCRIPTION
**Pull request by ANISH-GOTTAPU**
resolves #1488 

BGP Route Propagation test has GetAll which retrieves learnt prefixes. GetAll fails sometimes when BGP is taking sometime to exchange the routes.

Added WatchAll with timer to check if routes are learnt.

### CODE REVIEW

File: feature/bgp/addpath/otg_tests/route_propagation_test/route_propagation_test.go [hunk 1]
```diff
@@ -303,44 +303,56 @@ type OTGBGPPrefix struct {
 // }
 
 func checkOTGBGP4Prefix(t *testing.T, otg *otg.OTG, config gosnappi.Config, expectedOTGBGPPrefix OTGBGPPrefix) bool {
```
Comment (jasdeep-hundal) [inline feature/bgp/addpath/otg_tests/route_propagation_test/route_propagation_test.go:305]: Add t.Helper.

File: feature/bgp/addpath/otg_tests/route_propagation_test/route_propagation_test.go [hunk 2]
```diff
@@ -303,44 +303,56 @@ type OTGBGPPrefix struct {
...
+		for _, bgpPrefix := range bgpPrefixes {
+			if bgpPrefix.Address != nil && bgpPrefix.GetAddress() == expectedOTGBGPPrefix.Address &&
+				bgpPrefix.PrefixLength != nil && bgpPrefix.GetPrefixLength() == expectedOTGBGPPrefix.PrefixLength {
+				found = true
+				break
+			}
 		}
+		return found
+	} else {
+		t.Errorf("The BGP prefixes are not being learned.")
```
Comment (jasdeep-hundal) [inline feature/bgp/addpath/otg_tests/route_propagation_test/route_propagation_test.go:327]: Don't need t.Errorf here, error is reported in the main test function. Can also drop the else block and just have `return false`.

File: feature/bgp/addpath/otg_tests/route_propagation_test/route_propagation_test.go [hunk 3]
```diff
@@ -303,44 +303,56 @@ type OTGBGPPrefix struct {
...
+		return false
 	}
-	return found
 }
 
 func checkOTGBGP6Prefix(t *testing.T, otg *otg.OTG, config gosnappi.Config, expectedOTGBGPPrefix OTGBGPPrefix) bool {
```
Comment (jasdeep-hundal) [inline feature/bgp/addpath/otg_tests/route_propagation_test/route_propagation_test.go:332]: Add t.helper

File: feature/bgp/addpath/otg_tests/route_propagation_test/route_propagation_test.go [hunk 4]
```diff
@@ -303,44 +303,56 @@ type OTGBGPPrefix struct {
...
+			if bgpPrefix.Address != nil && bgpPrefix.GetAddress() == expectedOTGBGPPrefix.Address &&
+				bgpPrefix.PrefixLength != nil && bgpPrefix.GetPrefixLength() == expectedOTGBGPPrefix.PrefixLength {
+				found = true
+				break
+			}
 		}
-		time.Sleep(500 * time.Millisecond)
+		return found
+	} else {
+		t.Errorf("The BGP prefixes are not being learned.")
```
Comment (jasdeep-hundal) [inline feature/bgp/addpath/otg_tests/route_propagation_test/route_propagation_test.go:354]: Don't need t.Errorf here, error is reported in the main test function. Can also drop the else block and just have `return false`.

File: feature/bgp/addpath/otg_tests/route_propagation_test/route_propagation_test.go [hunk 5]
```diff
@@ -303,44 +303,56 @@ type OTGBGPPrefix struct {
 // }
```
Comment (jasdeep-hundal) [inline feature/bgp/addpath/otg_tests/route_propagation_test/route_propagation_test.go:303]: Can remove above TODO/commented-out function.

**Review by jasdeep-hundal — CHANGES_REQUESTED** [review]

**Review by thesrinath — APPROVED** [review]

**Review by jasdeep-hundal — APPROVED** [review]

**Pull request merged**